src/main/java/org/javacord/bot/commands/WikiCommand.java

```diff
@@ -2,28 +2,165 @@
-                .setThumbnail(getClass().getClassLoader().getResourceAsStream("javacord3_icon.png"), "png")
-                .setColor(Constants.JAVACORD_ORANGE);
-        channel.sendMessage(embed).join();
+    public void onCommand(DiscordApi api, TextChannel channel, String[] args) throws IOException {
+        try {
+            if (args.length == 0) { // Just an overview
+                EmbedBuilder embed = new EmbedBuilder()
+                        .setTitle("Javacord Wiki")
+                        .setDescription("The [Javacord Wiki](" + WikiParser.BASE_URL + "/wiki) is an excellent "
+                                + "resource to get you started with Javacord.\n")
+                        .addInlineField("Hint", "You can search the wiki using `!wiki [title|full] <search>")
+                        .setThumbnail(getClass().getClassLoader().getResourceAsStream("javacord3_icon.png"), "png")
+                        .setColor(Constants.JAVACORD_ORANGE);
+                channel.sendMessage(embed).join();
+            } else {
+                EmbedBuilder embed = new EmbedBuilder()
+                        .setThumbnail(getClass().getClassLoader().getResourceAsStream("javacord3_icon.png"), "png")
+                        .setColor(Constants.JAVACORD_ORANGE);
+                String searchString = String.join(" ", Arrays.copyOfRange(args, 1, args.length)).toLowerCase();
+                switch (args[0]) {
+                    case "page":
+                    case "p":
+                    case "title":
+                    case "t":
+                        populatePages(api, embed, titleOnly(searchString));
+                        break;
+                    case "full":
+                    case "f":
+                    case "content":
+                    case "c":
+                        populatePages(api, embed, fullSearch(searchString));
+                        break;
+                    default:
+                        searchString = String.join(" ", Arrays.copyOfRange(args, 0, args.length)).toLowerCase();
+                        populatePages(api, embed, defaultSearch(searchString));
+                }
+                channel.sendMessage(embed).join();
+            }
+        } catch (Throwable t) {
+            channel.sendMessage("Something went wrong: ```" + ExceptionLogger.unwrapThrowable(t).getMessage() + "```")
+                    .join();
+            // Throw the caught exception again. The sdcf4j will log it.
+            throw t;
+        }
+    }
+
+    private Predicate<WikiPage> defaultSearch(String searchString) {
+        return titleOnly(searchString).or(keywordsOnly(searchString));
+    }
+
+    private Predicate<WikiPage> fullSearch(String searchString) {
+        return titleOnly(searchString).or(keywordsOnly(searchString)).or(contentOnly(searchString));
+    }
+
+    private Predicate<WikiPage> titleOnly(String searchString) {
+        return p -> p.getTitle().toLowerCase().contains(searchString);
```

Comment:
Oh, one last nitpick maybe.
Single letter variable names are so absolutely meaningless.
I'd suggest to expand the `p`s to `page` and the `k` to `keyword`